algorithms.py

```diff
@@ -1148,6 +1159,20 @@ def print_heaps(self, L):
                 u = u.next
             print(s)
 
+    def census_event(self, time):
+        for pop in self.P:
+            # pop.print_state()
+            for ancestor in pop.iter_ancestors():
+                seg = ancestor
+                while seg is not None:
+                    self.flush_edges()
```

Comment:
I think you're flushing edges too much --- just needs to be done once per ancestor.